cirq-superstaq/cirq_superstaq/validation.py

```diff
@@ -4,10 +4,46 @@
 
 import cirq
 
+SUPPORTED_QID_TYPES = (
+    cirq.LineQubit,
+    cirq.LineQid,
+    cirq.GridQubit,
+    cirq.GridQid,
+    cirq.NamedQubit,
+    cirq.NamedQid,
+)
+
+
+def validate_qubit_types(circuits: object) -> None:
```

Comment:
```suggestion
def validate_qubit_types(circuits: cirq.Circuit | Sequence[cirq.Circuit]) -> None:
```
I think we should set the type as `circuits: cirq.Circuit | Sequence[cirq.Circuits]` since the function implicitly assumes this (otherwise the `circuit.all_qubits()` call would raise an attribute error)

Given that within the `validate_cirq_circuits()` function below ` validate_qubit_types()` is called after the circuits are validated we should be safe to enforce stricter typing

Reply:
That is indeed what I started with but for some silly reason stuck with `object` since mypy complained. However, it all checks out now with the current implementation -- so fixed, thanks!